### ISSUE TITLE
showLatestVersion doesn't appear to be working as intended in one scenario

### ISSUE THREAD

**Issue by amoeba** [NCEAS/metacatui#1697]
**Describe the bug**

Our "A newer version of this dataset exists..." banner doesn't work quite right when the next version from the current version is not readable by the client.

**To Reproduce**

0. Be not logged in
1. Go to https://arcticdata.io/catalog/view/urn%3Auuid%3A93472f4e-a491-48a8-b586-a63d9580685b
2. See the "A newer version..." banner flash and disappear

If the above link changes, you should be able to reproduce with a version chain that goes public->private->public.

**Expected behavior**

This view has enough information to tell the user that there is a new version, and even provide a link to the newer version so what's going on here isn't totally clear. Flashing the banner for a split second isn't desirable and we definitely want the user to know there's a new version even if they can't view it.

**Screenshots**

![2021-03-23 8 56 45](https://user-images.githubusercontent.com/563/112188945-57e80980-8bb8-11eb-988a-b7335b9e5a69.gif)

**Additional context**

**Comment by laurenwalker** [comment]
I agree that the banner flashing is not desirable. The banner shows up because sometimes it takes so long to traverse a version chain,[ that we implemented a 'loading' version of the message while it is doing so. ](https://github.com/NCEAS/metacatui/issues/543) But it sounds like the implementation needs some finessing so it doesn't show up then disappear.

I'm not so sure we want to display a link to a newer version of the dataset if it is not accessible by the person looking at the dataset. I'm not sure how useful it is to know that a newer version exists if you can never see it. Perhaps a less obvious message for these scenarios.

**Comment by amoeba** [comment]
I think knowing there's a new version but a version you can't access is probably only marginally useful. Maybe more frustrating than useful. Maybe the display of the banner could be delayed until a new version is found we can read?

**Comment by laurenwalker** [comment]
We could wait to show the loading banner until at least one more accessible version is found. So even if there are dozens or newer versions, we know at least one can be linked to before showing it.

**Comment by amoeba** [comment]
That sounds good to me.

**Comment by robyngit** [comment]
With #2567 the viewer no longer shows a newer version banner when the next version is private. A `public A → private B` check therefore leaves the banner hidden, fixing the main concern in this issue.

However, in the `public A → private B → public C` case, version chain traversal stops at B because it cannot read B's sys meta, so it cannot discover C through that chain. I think that's a fundamental limitation of the versioning system, so not something we can fix in MetacatUI.

**Comment by mbjones** [comment]
I agree that is a limit of the current versioning model. That said, we've been wanting to improve the Metacat/DataONE versioning model, particularly by adding a dedicated version chain query API with two new methods:

```java
    listVersions(id, range="prior|next", count=5) : VersionList
    getVersion(id, position="first|prior|next|last") : Identifier
```

See extensive discussion in https://github.com/NCEAS/metacat/issues/1429#issuecomment-3212511266 which essentially described the core of the implementation.

We could decide that version information through these APIs itself is not protected, while `read`  access for `/object/PID` and `/meta/PID` still could be protected. This would allow clients to much more intelligently navigate version chains, and your version page wouldn't have to walk the whole version chain to construct the history -- it could do it in a single API call to `listVersions`.  Please do comment on that Metacat issue with any changes you see as helpful -- I'd like to see if @taojing2002 can fold this new API into a Metacat release.
